web/libs/editor/src/components/KonvaVector/KonvaVector.tsx

```diff
@@ -0,0 +1,1606 @@
+          if (transformerRef.current) {
+            // Set up proxy nodes once - transformer will manage them independently
+            // Use getAllPoints() to get the correct proxy nodes for all points
+            const allPoints = getAllPoints();
+            const nodes = Array.from(selectedPoints)
+              .map((index) => {
+                // Ensure the index is within bounds of all points
+                if (index < allPoints.length) {
+                  return proxyRefs.current[index];
+                }
+                return null;
+              })
+              .filter((node) => node?.getAbsoluteTransform) as Konva.Node[];
+
+            if (nodes.length > 0) {
+              // Always set the complete set of nodes - transformer will handle positioning
+              transformerRef.current.nodes(nodes);
+              transformerRef.current.getLayer()?.batchDraw();
+            }
+          }
+        }, 0);
+      } else {
+        // Clear transformer when selection is less than 2 points
+        setTimeout(() => {
+          if (transformerRef.current) {
+            transformerRef.current.nodes([]);
+            transformerRef.current.getLayer()?.batchDraw();
+          }
+        }, 10);
+      }
+    }
+  }, [selectedPoints]); // Only depend on selectedPoints, not initialPoints
+
+  // Note: We don't update proxy node positions during transformation
+  // The transformer handles positioning the proxy nodes itself
+  // This prevents conflicts and maintains the transformer's rotation state
+
+  // Helper function to generate shape data and call transformation complete callback
+  const notifyTransformationComplete = () => {
+    // Get all points
+    const allPoints = getAllPoints();
+
+    if (format === "simple") {
+      // Export in simple format
+      // For simple format, we need to call a different callback or handle differently
+      // since onTransformationComplete expects the complex format
+    } else {
+      // Export in regular format
+      const exportedPoints = allPoints.map((point) => {
+        const controlPoints: Array<{ x: number; y: number }> = [];
+
+        if (point.isBezier) {
+          if (point.controlPoint1) {
+            controlPoints.push({
+              x: point.controlPoint1.x,
+              y: point.controlPoint1.y,
+            });
+          }
+          if (point.controlPoint2) {
+            controlPoints.push({
+              x: point.controlPoint2.x,
+              y: point.controlPoint2.y,
+            });
+          }
+        }
+
+        return {
+          x: point.x,
+          y: point.y,
+          bezier: point.isBezier || false,
+          controlPoints,
+        };
+      });
+
+      // Check if we have enough points based on minPoints constraint
+      const incomplete = minPoints !== undefined && allPoints.length < minPoints;
+
+      const shapeData = {
+        type: (allowClose ? "polygon" : "polyline") as "polygon" | "polyline",
```

Comment:
please create an enum or a type then update usage everywhere we use "polygon" or "polyline". this will help reduce the chance of mistakes relating to spelling plus help document the code a little bit better